core/src/main/java/io/kestra/core/docs/AbstractClassDocumentation.java

```diff
@@ -42,7 +46,11 @@ protected AbstractClassDocumentation(JsonSchemaGenerator jsonSchemaGenerator, Cl
             .filter(entry -> !entry.getKey().equals("io.kestra.core.models.tasks.Task"))
             .map(entry -> {
                 Map<String, Object> value = (Map<String, Object>) entry.getValue();
-                value.put("properties", flatten(properties(value), required(value)));
+                try {
+                    value.put("properties", flatten(properties(value), required(value), AbstractRetry.class.isAssignableFrom(Class.forName(entry.getKey()))));
```

Comment:
Required for class like Retry where they have a "type" property that we want to keep

Reply:
There may be multiple `type` that we will need to keep, better to extract that in a dedicated method like `isTypeToKeep(entry.getKey)` I also think that if there is a class related exception we should just goes with the default (so returning null)